x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/config/RuleScope.java

```diff
@@ -43,7 +44,7 @@
             Map<String, FilterRef> scope = new HashMap<>();
             for (Map.Entry<String, Object> entry : unparsedScope.entrySet()) {
                 try (XContentBuilder builder = XContentFactory.jsonBuilder()) {
-                    builder.map((Map<String, ?>) entry.getValue());
+                    builder.map((Map<String, Object>) entry.getValue());
```

Comment:
`builder.map` expects a `Map<String, ?>`, so I don't think we need to change this line.